File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.cpp
class Solution {
public:
    bool isVowel(char c){
        return c=='a' || c=='A' || c=='e' || c=='E' || c=='o' || c=='O' || c=='i' || c=='I' || c=='u' || c=='U';    
    }
    
    string reverseVowels(string s) {
        int start=0, end = s.size()-1;
        
        while(start < end){
            while(start < s.size()-1 && !isVowel(s[start])){
                start++;
            }
            
            while(end >= 0 && !isVowel(s[end])){
                end--;
            }
            
            if(start < end){
                swap(s[start++], s[end--]);
            }
        }
        return s;
    }
};

File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.py
class Solution:
    def reverseVowels(self, s: str) -> str:
        left, right = 0, len(s)-1
        s = list(s)
        val = ['a', 'e', 'i', 'o', 'u','A', 'E', 'O', 'I', 'U']
        
        while left < right:
            while left < len(s) and (s[left] not in val):
                left += 1
            
            while right >=0 and s[right] not in val:
                right -= 1
            
            if left < right:
                s[left], s[right] = s[right], s[left]
                left += 1
                right -= 1

        return ''.join(char for char in s)